Index: tests/test_4.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_4.py b/tests/test_4.py
new file mode 100644
--- /dev/null	(date 1669111043326)
+++ b/tests/test_4.py	(date 1669111043326)
@@ -0,0 +1,31 @@
+import pytest
+from hw_1.tasks.task_4 import check_sum_of_four
+
+
+@pytest.mark.parametrize(
+    ["value", "expected_result"],
+    [
+        (
+            [
+                [0, 1, 1, 2],
+                [4, 5, 6, 7],
+                [-2, 7, -8, 13],
+                [-2, 9, 1, -6]
+            ],
+            2,
+        ),
+        (
+            [
+                [-3, 7, 4],
+                [-2, 5, 1],
+                [11, 6, 3],
+                [9, -18, -5]
+            ],
+            2,
+        )
+    ]
+)
+def test_check_sum_of_four(value: list, expected_result: int):
+    actual_result = check_sum_of_four(value)
+
+    assert actual_result == expected_result
Index: ../requirements.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/../requirements.txt b/../requirements.txt
new file mode 100644
--- /dev/null	(date 1669111043310)
+++ b/../requirements.txt	(date 1669111043310)
@@ -0,0 +1,2 @@
+pytest
+black
\ No newline at end of file
Index: tasks/task_4.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tasks/task_4.py b/tasks/task_4.py
new file mode 100644
--- /dev/null	(date 1669111043310)
+++ b/tasks/task_4.py	(date 1669111043310)
@@ -0,0 +1,9 @@
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:
+    tup_count = 0
+    for i in range(len(a)):
+        if a[i] + b[i] + c[i] + d[i] == 0:
+            tup_count += 1
+    return tup_count
Index: tests/test_3.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import pytest\r\nfrom hw_1.tasks.task_3 import find_maximum_and_minimum\r\n\r\n\r\n@pytest.mark.parametrize(\r\n    [\"value\", \"expected_result\"],\r\n    [\r\n        ('file_name.txt', [1, 6]),\r\n    ],\r\n)\r\ndef test_find_maximum_and_minimum(value: str, expected_result: tuple[int, int]):\r\n    actual_result = find_maximum_and_minimum(value)\r\n\r\n    assert actual_result == expected_result\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_3.py b/tests/test_3.py
--- a/tests/test_3.py	(revision 79a0ef4a78585af68f346f013ccad717d5a14d1a)
+++ b/tests/test_3.py	(date 1669111043295)
@@ -5,7 +5,7 @@
 @pytest.mark.parametrize(
     ["value", "expected_result"],
     [
-        ('file_name.txt', [1, 6]),
+        ('C:\LoremAmoris\PyProjects\hw_1\\tests\\file_name.txt', [1, 6]),
     ],
 )
 def test_find_maximum_and_minimum(value: str, expected_result: tuple[int, int]):
